conda/signature.py

```diff
@@ -0,0 +1,66 @@
+import sys
+import base64
+import hashlib
+from os.path import abspath, expanduser, join
+
+from conda.compat import PY3
+
+from Crypto.PublicKey import RSA
+
+
+KEYS = {}
+KEYS_DIR = abspath(expanduser('~/.conda/keys'))
+
+
+def sig2ascii(i):
+    ret = []
+    while i:
+        i, r = divmod(i, 256)
+        ret.append(r)
+    if PY3:
+        s = bytes(n for n in ret[::-1])
+    else:
+        s = ''.join(chr(n) for n in ret[::-1])
+    return base64.b64encode(s).decode('utf-8')
+
+
+def ascii2sig(s):
+    res = 0
+    for c in base64.b64decode(s):
+        res *= 256
+        res += (c if PY3 else ord(c))
+    return res
+
+
+def hash_file(path):
+    h = hashlib.new('sha256')
+    with open(path, 'rb') as fi:
+        while True:
+            chunk = fi.read(262144)
```

Comment:
This is actually not a magic number, it is just 2^18, ie. 256KB.  I added a comment.